src/registry/number/edit.tsx

```diff
@@ -0,0 +1,221 @@
+import {NumberComponentSchema} from '@open-formulieren/types';
+import {useFormikContext} from 'formik';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {
+  BuilderTabs,
+  ClearOnHide,
+  Description,
+  Hidden,
+  IsSensitiveData,
+  Key,
+  Label,
+  PresentationConfig,
+  Registration,
+  SimpleConditional,
+  Suffix,
+  Tooltip,
+  Translations,
+  Validate,
+  useDeriveComponentKey,
+} from '@/components/builder';
+import {Checkbox, NumberField, TabList, TabPanel, Tabs} from '@/components/formio';
+import {getErrorNames} from '@/utils/errors';
+
+import {EditFormDefinition} from '../types';
+
+/**
+ * Form to configure a Formio 'number' type component.
+ */
+const EditForm: EditFormDefinition<NumberComponentSchema> = () => {
+  const [isKeyManuallySetRef, generatedKey] = useDeriveComponentKey();
+  const {errors} = useFormikContext<NumberComponentSchema>();
+
+  const erroredFields = Object.keys(errors).length
+    ? getErrorNames<NumberComponentSchema>(errors)
+    : [];
+  // TODO: pattern match instead of just string inclusion?
+  // TODO: move into more generically usuable utility when we implement other component
```

Comment:
We'll know what we need to be able to express. Defining the types, is a bit like unittests, you ultimately I want them to help me make sense of the code and aid in changing the code with trust, not prevent me from doing so.

And aid can come in the form of pointing out all the locations I need to refactor when changing the type.

Having opaque types, can help with that. So we should be careful about which types should end up in that types package, and which will only prevent us from change... Change is inevitable. All we do should help with change.

Reply:
This TODO was not about the types package, but internally in the builder library!

The intent of the types package is to model the real world:

* strongly type any extensions we add to the formio schemas, which can be relevant for builder, renderer *and even* the backend (e.g. the date `openForms.minDate` configuration in the builder gets transformed by the backend to `datePicker.minDate`, so this requires `openForms.minDate` typing for the builder (and backend) and `datePicker.minDate` typing for the renderer & backend.
* remove the bits from formio that don't apply to our product(s) - this aids developers who look at formio and implement things in preventing them from implementing things that will never be executed. That was something I saw happening a lot when Sven was working on the renderer

It does create some more hurdles - if we implement or change functionality in our backend/frontend, this means that the types package must be updated first to model this, but I don't think it will be that problematic since it's mostly in the "concrete" domain (as opposed to the domain of generics).